Fix rebase status displayed during a rebase

pmiossec opened the pull request:
Fixes #8540, #7152 and #6350

## Proposed changes

- Fix the list of displayed commits that are rebased to always display all of them and with the good `Status` displayed
- other adjustments:
    * Move the important 'Status' column to be the first one (and not lost after author and date so that it could be more easily compare to the `Action` and `Subject`)
    * Display commit hash
    * Rename "Name" header to "Action" (because it display the define action. Especially when doing an interactive rebase)
    * Being able to open a commit diff window by double click (like it is possible to see the patch file for the patches)
    * Set a minimum height for a row (to avoid possible strange behavior)
    * Resize some columns



## Screenshots <!-- Remove this section if PR does not change UI -->

### Before

![image](https://user-images.githubusercontent.com/460196/95751244-9fc84380-0c9e-11eb-9635-1b70503950c0.png)

### After

* New list:
![image](https://user-images.githubusercontent.com/460196/100289840-29e01900-2f7a-11eb-84c8-5db939e59d23.png)

* Opening a commit:
![image](https://user-images.githubusercontent.com/460196/100290463-b5a67500-2f7b-11eb-9f5c-7f545337d0f8.png)



## Test methodology <!-- How did you ensure quality? -->

- Manual


## Test environment(s) <!-- Remove any that don't apply -->

- GIT <!-- Add version 2.11 or above -->
- Windows <!-- Add version 7 SP1 or above -->

<!-- Mention language, UI scaling, or anything else that might be relevant -->


----

:black_nib: I contribute this code under [The Developer Certificate of Origin](../blob/master/contributors.txt).

ivangrek commented on GitCommands/Git/GitModule.cs:
```diff
@@ -1926,14 +1926,14 @@ private string GetNextRebasePatch()
-        {
-            return File.Exists(GetRebaseDir() + "git-rebase-todo");
-        }
+        private string RebaseTodoFilePath => GetRebaseDir() + "git-rebase-todo.backup";
+        private string CurrentFilePath => GetRebaseDir() + "stopped-sha";
```
Could we use constants for strings?

RussKie replied:
In which way?

gerhardol replied:
In this case the string is used only once (related to this routine) and the statement is is the accessor so it is used as a constant with a dynamic part.
I lean on not having a constant here, as it adds another layer. (Me and @ivangrek has had that discussion in #8628.) At the same time I see the point when seeing a stray string in code.

In a wider usage, "git-rebase-todo" is hardcoded also in GitUI\Editor\FileViewerInternal.cs SetHighlightingForFile() to highlight in the editor. This file does not need to be syntax highlighted what I see.

To use constants for files GE uses should be a TODO.

ivangrek commented on GitCommands/Git/GitModule.cs:
```diff
@@ -1951,19 +1953,28 @@ public IReadOnlyList<PatchFile> GetInteractiveRebasePatchFiles()
+                    string commitHash = parts[1];
+                    CommitData data = _commitDataManager.GetCommitData(commitHash, out var error);
+                    var isApplying = currentCommitShortHash != null && commitHash.StartsWith(currentCommitShortHash);
+                    if (!isCurrentFound && isApplying)
+                    {
+                        isCurrentFound = isApplying;
                     }
```
``` suggestion
   isCurrentFound |= isApplying
```

ivangrek replied:
false false => false
false true => true
true false => true
true true => true

OR

gerhardol reviewed (approved):
Great change
Also a step towards #5161?

gerhardol commented:
This should be included, I added to next release milestone

pmiossec commented:
Updated to better handle skipped commits/patches. Now, if the popup is closed and opened again, the skipped commit are still displayed as "Skipped" (instead of "Applied").

And better highlight the currently applied commit/patch (with conflicts) with some color:
![image](https://user-images.githubusercontent.com/460196/101412594-2a19d600-38e3-11eb-9eb5-3c7522bc5a8d.png)

And fixed a bug so that the currently applied patch line was not selected (due to the selecting was not made in the UI thread).

RussKie commented:
@msftbot merge in 24 hours

jrfnl commented:
@pmiossec Thanks so much! This has been bugging me for quite a while, but never got round to opening an issue. Awesome to see this fixed!
